tools/autograd/gen_variable_type.py

```diff
@@ -97,11 +97,15 @@
 }
 """)
 
-DERIVATIVE_MULTI = CodeTemplate("""\
-if (should_compute_output({ ${idxs} })) {
+GRAD_INPUT_MASK = CodeTemplate("""\
   auto grad_input_mask = std::array<bool, ${n}>{
     ${masks}
-  };
+  };\
+""")
+
+DERIVATIVE_MULTI = CodeTemplate("""\
+if (should_compute_output({ ${idxs} })) {
+${grad_input_mask}
```

Comment:
Can you fix the indentation here? indent `${grad_input_mask}` but not:

```
auto grad_input_mask = ....
```

I think the `code_template` is smart enough that the output will have the proper indentation.

Reply:
If I do this, it ends up looking like this:
```cpp
if (should_compute_output({...}) {
  auto grad_input_mask = std::array<bool, 3>{
  should_compute_output(...),
  should_compute_output(...),
  should_compute_output(...),
};
  <call here>
}
```

That's the only way I've found to make it indent this part properly.

Reply:
Oh, that's too bad. I had thought the `code_template.py` handled this case, but I guess it only does the first line.

Reply:
I can try to fix that if you prefer that

Reply:
It'd be nice if `code_template.py` handled it, but there's no need to block this PR